scan-payment/src/main/java/com/getbouncer/scan/payment/analyzer/NameDetectAnalyzer.kt

```diff
@@ -73,7 +105,31 @@ class NameDetectAnalyzer private constructor(
         )
     }
 
-    private suspend fun processPredictions(
+    private suspend fun processExpiryPrediction(
+        expRect: RectF,
+        bitmapForObjectDetection: Bitmap
+    ): ExpiryDetect.Expiry? {
+        // Crop out the expected aspect ratio
+        val targetAspectRatio = ASPECT_RATIO
+        val scaledRect = expRect.scaled(bitmapForObjectDetection.size())
+        val scaledExpRectNewHeight = scaledRect.width() * targetAspectRatio
+
+        val rect = Rect(
+            scaledRect.left.toInt(),
+            (scaledRect.centerY() - scaledExpRectNewHeight / 2).toInt(),
+            scaledRect.right.toInt(),
+            (scaledRect.centerY() + scaledExpRectNewHeight / 2).toInt()
+        )
```

Comment:
```suggestion
        val rect = ASPECT_RATIO.scaleAndCenterWithin(bitmapForObjectDetection.size())
```

Reply:
i think this is different - what i'm trying to do here is this: given an expiry rectF and the target aspect ratio that this model expects, return a Rect that respect this aspect ratio, making sure the width is fixed and the center remains the same

My understanding of `scaleAndCenterWithin` is returning the largest rect of the given aspect ratio in the big Rect

Reply:
got it, I misunderstood. This makes sense.